include/openthread/dns_client.h

```diff
@@ -220,21 +277,25 @@ typedef struct otDnsServiceInfo
  *
  * This function is available when `OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE` is enabled.
  *
+ * The @p aConfig can be NULL. In this case the default config (from `otDnsClientGetDefaultConfig()`) will be used as
+ * the config for this query. In a non-NULL @p aConfig, some of the fields can be left unspecified (value zero). The
+ * unspecified fields are then replaced by the values from the default config.
+ *
  * @param[in]  aInstance        A pointer to an OpenThread instance.
- * @param[in]  aServerSockAddr  A pointer to the server socket address.
  * @param[in]  aServiceName     The service name to query for (MUST NOT be NULL).
  * @param[in]  aCallback        A function pointer that shall be called on response reception or time-out.
  * @param[in]  aContext         A pointer to arbitrary context information.
+ * @param[in]  aConfig          A pointer to the config to use for this query.
  *
  * @retval OT_ERROR_NONE        Query sent successfully. @p aCallback will be invoked to report the status.
  * @retval OT_ERROR_NO_BUFS     Insufficient buffer to prepare and send query.
  *
  */
-otError otDnsClientBrowse(otInstance *        aInstance,
-                          const otSockAddr *  aServerSockAddr,
-                          const char *        aServiceName,
-                          otDnsBrowseCallback aCallback,
-                          void *              aContext);
+otError otDnsClientBrowse(otInstance *            aInstance,
+                          const char *            aServiceName,
+                          otDnsBrowseCallback     aCallback,
+                          void *                  aContext,
+                          const otDnsQueryConfig *aConfig);
```

Comment:
```suggestion
);
```

Remove this argument since the caller can already do everything it want with `otDnsClientSetDefaultConfig`.
I think we should make the API as simple as possible, unless the benefits are clear?

Reply:
Answered above.